lisa/core/tool.py

```diff
@@ -0,0 +1,96 @@
+from __future__ import annotations
+
+import pathlib
+from abc import ABC, abstractmethod
+from typing import TYPE_CHECKING, List, Optional, TypeVar
+
+from lisa.util.executableResult import ExecutableResult
+from lisa.util.process import Process
+
+if TYPE_CHECKING:
+    from lisa.core.node import Node
+
+T = TypeVar("T")
+
+
+class Tool(ABC):
+    def __init__(self, node: Node) -> None:
+        self.node: Node = node
+        self._isInstalled: Optional[bool] = None
+        self.initialize()
+
+    def initialize(self) -> None:
+        pass
+
+    @property
+    def dependentedTools(self) -> List[T]:
```

Comment:
Ah, we'll want to call this something other than `dependentedTools` 😅 maybe just `dependencies`?

Reply:
sure, feel free to correct my English. 😄